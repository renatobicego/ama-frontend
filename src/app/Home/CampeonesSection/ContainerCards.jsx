import CampeonCard from "./CampeonCard"


const ContainerCards = ({campeones}) => {
    return(
        <div className="w-full flex justify-between py-6 px-16">
            {campeones.map((campeon, i) => <CampeonCard campeon={campeon} key={i}/>)}
        </div>
    )
}

export default ContainerCards