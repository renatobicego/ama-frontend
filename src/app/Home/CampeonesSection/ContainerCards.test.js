import {cleanup, prettyDOM, render, screen } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import ContainerCards from "./ContainerCards"


describe('<ContainerCards />', () => {

    let containerCampeones 

    const mockedCampeones = [
        {
            name: 'Renzo Cremaschi',
            imgHref: '/test/test.jpg',
            pruebas: ['110 C/V', '200m']
        },
    ]

    // Render component
    beforeEach(() => {
        containerCampeones  = render(<ContainerCards campeones={mockedCampeones}/>)
    })

    afterEach(() => {
        cleanup()
    })

    test('renders content', () => {
        expect(containerCampeones .container).toMatchSnapshot()

        // Check img
        const img = screen.getByAltText(`Foto de ${mockedCampeones[0].name}`)
        expect(img).toBeInTheDocument()

        // String of pruebas 
        let pruebasText = mockedCampeones[0].pruebas.map(prueba => prueba + ' / ')

        // Check pruebas rendered
        const pruebas = screen.getByText(pruebasText.join('').trim())
        expect(pruebas).toBeInTheDocument()
    })
})