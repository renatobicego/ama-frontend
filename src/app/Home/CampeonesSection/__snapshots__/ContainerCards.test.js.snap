// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerCards /> renders content 1`] = `
<div>
  <div
    class="w-full flex justify-evenly py-6 px-16"
  >
    <div
      class="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md mt-6 w-95% md:w-[45%] lg:w-[30%] overflow-hidden"
    >
      <div
        class="bg-clip-border overflow-hidden bg-white text-gray-700 relative w-full m-0 rounded-none"
      >
        <img
          alt="Foto de Renzo Cremaschi"
          class="h-44 md:h-52 lg:h-64 object-cover"
          data-nimg="1"
          decoding="async"
          height="700"
          loading="lazy"
          src="/_next/image?url=%2Ftest%2Ftest.jpg&w=1920&q=75"
          srcset="/_next/image?url=%2Ftest%2Ftest.jpg&w=750&q=75 1x, /_next/image?url=%2Ftest%2Ftest.jpg&w=1920&q=75 2x"
          style="color: transparent;"
          width="700"
        />
      </div>
      <div
        class="p-6"
      >
        <h5
          class="block antialiased tracking-normal text-xl font-semibold leading-snug mb-2 text-title font-title"
        >
          Renzo Cremaschi
        </h5>
        <p
          class="block antialiased text-base font-light leading-relaxed text-title font-title"
        >
          110 C/V / 
          200m / 
        </p>
      </div>
    </div>
  </div>
</div>
`;
